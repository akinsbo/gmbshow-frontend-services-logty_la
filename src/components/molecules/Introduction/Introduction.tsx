import * as React from 'react';

const Introduction = (props: { children: React.ReactNode }) => {
    const { children } = props;

    return (
        <div>
            <h2 aria-label= "introduction"
                aria-role="heading">{children}</h2>
        </div>
    );
}

export default Introduction