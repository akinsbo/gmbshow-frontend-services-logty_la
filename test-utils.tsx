import React from 'react'
import { render, queries } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/styles'
import { createMuiTheme } from '@material-ui/core/styles'
import { I18nextProvider } from 'react-i18next'
import defaultStrings from './i18n/en-x-default'
import * as customQueries from './custom-queries'
import grey from '@material-ui/core/colors/grey'
import orange from '@material-ui/core/colors/orange'
import purple from '@material-ui/core/colors/purple'
import yellow from '@material-ui/core/colors/yellow'
// A theme with custom primary and secondary color.
// It's optional.
const theme = createMuiTheme({
    palette: {
        primary: {
            dark: purple[900],
            light: orange[500],
            main: grey[700]
        },
        secondary: {
            dark: yellow[700],
            light: purple[300],
            main: yellow[500]
        },
    }
})

const AllTheProviders = ({ children }: any) =>
    <ThemeProvider theme={theme} >
        <I18nextProvider i18n={defaultStrings} >
            {children}
        </I18nextProvider>
    </ThemeProvider >


const customRender = (ui: any, options = {}) =>
    render(ui, { wrapper: AllTheProviders, queries: { ...queries, ...customQueries }, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }