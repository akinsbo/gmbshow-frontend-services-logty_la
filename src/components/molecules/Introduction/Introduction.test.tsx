import * as React from "react"
import { render, screen } from '../../../../test-utils'
import Introduction from "./Introduction"

describe("Introduction", () => {
  it("renders correctly", () => {
    const container = render(
      <Introduction>This is the Promise</Introduction>
    )
    expect(container).toMatchSnapshot()
    expect(screen.getByRole('heading', {name: /introduction/i})).toHaveTextContent("This is the Promise")
  })
})