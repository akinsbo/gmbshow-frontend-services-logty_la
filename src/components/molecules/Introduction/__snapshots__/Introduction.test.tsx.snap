// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Introduction renders correctly 1`] = `undefined`;
